### ISSUE TITLE
Remix 2.7.0 has an invalid default getLoadContext for Cloudflare Pages

### ISSUE THREAD

**Issue by AdiRishi** [remix-run/remix#8816]
### Reproduction

https://github.com/AdiRishi/remix-2.7.0-env-bug

### System Info

```shell
System:
  OS: macOS 14.3.1
  CPU: (8) arm64 Apple M1 Pro
  Memory: 6.61 GB / 32.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
  Yarn: 1.22.19 - ~/.nvm/versions/node/v20.11.0/bin/yarn
  npm: 10.4.0 - ~/.nvm/versions/node/v20.11.0/bin/npm
  pnpm: 8.15.3 - ~/.nvm/versions/node/v20.11.0/bin/pnpm
  bun: 1.0.25 - /opt/homebrew/bin/bun
Browsers:
  Chrome: 121.0.6167.184
  Safari: 17.3.1
npmPackages:
  @remix-run/cloudflare: ^2.7.0 => 2.7.0
  @remix-run/cloudflare-pages: ^2.7.0 => 2.7.0
  @remix-run/css-bundle: ^2.7.0 => 2.7.0
  @remix-run/dev: ^2.7.0 => 2.7.0
  @remix-run/react: ^2.7.0 => 2.7.0
```

### Used Package Manager

pnpm

### Expected Behavior

When accessing `args.context.env` from a loader we expect this to contain environment variables loaded from `.dev.vars`

### Actual Behavior

The `2.7.0` release seems to make the `getLoadContext` function optional in `createPagesFunctionHandler`. The expectation according to the release notes is that this will bring the implementation inline with the vite setup and correctly default to loading `context.env`.

However, this is not the case. The reasoning seems to be that the default implementation of `getLoadContext` is `(context) => ({ env: context })`. The correct implementation actually seems to be `({ context }) => ({ env: context.cloudflare.env })` since both the input arguments and context type seem to have changed in the latest release.

To be clear, this is what shows up in the `cloudflare-pages` template for `server.ts`
```ts
import { logDevReady } from "@remix-run/cloudflare";
import { createPagesFunctionHandler } from "@remix-run/cloudflare-pages";
import * as build from "@remix-run/dev/server-build";

if (process.env.NODE_ENV === "development") {
  logDevReady(build);
}

export const onRequest = createPagesFunctionHandler({ build });
```
A correct `server.ts` looks like this
```ts
import { logDevReady } from '@remix-run/cloudflare';
import { createPagesFunctionHandler } from '@remix-run/cloudflare-pages';
import * as build from '@remix-run/dev/server-build';

if (process.env.NODE_ENV === 'development') {
  logDevReady(build);
}

export const onRequest = createPagesFunctionHandler({
  build,
  getLoadContext: ({ context }) => ({ env: context.cloudflare.env }),
});
```

I've demonstrated this in the reproduction repository as well.

**Comment by markdalgleish** [comment]
Fixed in https://github.com/remix-run/remix/pull/8819, released in v2.7.1.

**Comment by AdiRishi** [comment]
Awesome!

Just a question, I see in the PR that you've marked the old args passed to `getLoadContext` as deprecated. To maintain compatibility with v3 going forward, is it expected that we use `getLoadContext: ({ context }) => ({ env: context.cloudflare.env })`?
